mne_sandbox/connectivity/cfc.py

```diff
@@ -121,17 +122,136 @@ def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+    f_amp : array, dtype float, shape (n_bands_amp, 2,)
+        The frequency ranges to use for the phase-modulated amplitude.
+        PAC will be calculated between n_bands_phase * n_bands_amp frequencies.
+    ixs : array-like, shape (n_ch_pairs x 2)
+        The indices for low/high frequency channels. PAC will be estimated
+        between n_ch_pairs of channels. Indices correspond to rows of `data`.
+    sfreq : float
+        The sampling frequency of the data.
+    times : array, shape (n_times,) | None
+        An array of times corresponding to the final axis of inputs. Will be
+        used in conjunction with tmin/tmax if `events` is None.
+    pac_func : {'plv', 'glm', 'mi_canolty', 'mi_tort', 'ozkurt'}
+        The function for estimating PAC. Corresponds to functions in
+        `pacpy.pac`. Defaults to 'ozkurt'. If multiple frequency bands are used
+        then `plv` cannot be calculated.
+    tmin : float | list of floats, shape (n_pac_windows,) | None
+        If `events` is not provided, it is the start time to use in `inst`.
+        If `events` is provided, it is the time (in seconds) to include before
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `min(inst.times)`.
+    tmax : float | list of floats, shape (n_pac_windows,) | None
+        If `events` is not provided, it is the stop time to use in `inst`.
+        If `events` is provided, it is the time (in seconds) to include after
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `max(inst.n_times)`.
+    n_cycles_ph : float, int | array of floats, shape (n_bands_phase,)
+        The number of cycles to be included in the window for each band-pass
+        filter for phase. Defaults to 3.
+    n_cycles_am : float, int | array of floats, shape (n_bands_amp,)
+        The number of cycles to be included in the window for each band-pass
+        filter for amplitude. Defaults to 3.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., `sklearn.preprocessing.scale`.
+        Defaults to no scaling.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    """
+    def __init__(self, f_phase, f_amp, ixs, sfreq, times,
+                 pac_func='ozkurt', tmin=0., tmax=1., n_cycles_ph=3,
+                 n_cycles_am=3, scale_amp_func=None, n_jobs=1):
+        events, times, tmin, tmax = _check_pac_times(None, times, tmin, tmax)
+        (ixs, f_phase, f_amp, n_cycles_ph,
+         n_cycles_am, pac_func) = _check_pac_data(
+            ixs, f_phase, f_amp, n_cycles_ph, n_cycles_am, pac_func)
+
+        self.f_phase = f_phase
+        self.f_amp = f_amp
+        self.tmin = tmin
+        self.tmax = tmax
```

Comment:
The tmin / tmax parameters let you supply multiple timepoints for PAC calculation, and yeah they also avoid the problem of edge artifacts. E.g., you'd want to supply Epochs w/ times.min() = -1, and times.max() = 1, and then supply tmin/tmax of `[0, .1]` or something. Otherwise we might get edge artifacts...also the pacpy filtering code is a little wonky and has a tendency to return exceptions because of window length issues.